src/openms/source/FORMAT/ChromeleonFile.cpp

```diff
@@ -0,0 +1,131 @@
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/FORMAT/ChromeleonFile.h>
+
+namespace OpenMS
+{
+  ChromeleonFile::ChromeleonFile() {}
+
+  ChromeleonFile::~ChromeleonFile() {}
+
+  void ChromeleonFile::load(const String& filename, MSExperiment& experiment) const
+  {
+    std::ifstream ifs(filename, std::ifstream::in);
+    if (!ifs.is_open())
+    {
+      throw Exception::FileNotFound(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
+    }
+    const Size BUFSIZE = 65536;
+    char line[BUFSIZE];
+    experiment.clear(true);
+    MSChromatogram chromatogram;
+    std::cmatch m;
+    std::regex re_injection("^Injection\t(.+)", std::regex::optimize);
+    std::regex re_processing_method("^Processing Method\t(.+)", std::regex::optimize);
+    std::regex re_instrument_method("^Instrument Method\t(.+)", std::regex::optimize);
+    std::regex re_injection_date("^Injection Date\t(.+)", std::regex::optimize);
+    std::regex re_injection_time("^Injection Time\t(.+)", std::regex::optimize);
+    std::regex re_detector("^Detector\t(.+)", std::regex::optimize);
+    std::regex re_signal_quantity("^Signal Quantity\t(.+)", std::regex::optimize);
+    std::regex re_signal_unit("^Signal Unit\t(.+)", std::regex::optimize);
+    std::regex re_signal_info("^Signal Info\t(.+)", std::regex::optimize);
+    std::regex re_raw_data("^Raw Data:", std::regex::optimize);
+    while (!ifs.eof())
+    {
+      ifs.getline(line, BUFSIZE);
+      if (std::regex_search(line, m, re_injection))
+      {
+        experiment.setMetaValue("mzml_id", std::string(m[1]));
+      }
+      else if (std::regex_search(line, m, re_processing_method))
+      {
+        experiment.getExperimentalSettings().getInstrument().getSoftware().setName(String(m[1]));
+      }
+      else if (std::regex_search(line, m, re_instrument_method))
+      {
+        experiment.getExperimentalSettings().getInstrument().setName(String(m[1]));
+      }
+      else if (std::regex_search(line, m, re_injection_date))
+      {
+        experiment.setMetaValue("injection_date", std::string(m[1]));
+      }
+      else if (std::regex_search(line, m, re_injection_time))
+      {
+        experiment.setMetaValue("injection_time", std::string(m[1]));
+      }
+      else if (std::regex_search(line, m, re_detector))
+      {
+        experiment.setMetaValue("detector", std::string(m[1]));
+      }
+      else if (std::regex_search(line, m, re_signal_quantity))
+      {
+        experiment.setMetaValue("signal_quantity", std::string(m[1]));
+      }
+      else if (std::regex_search(line, m, re_signal_unit))
+      {
+        experiment.setMetaValue("signal_unit", std::string(m[1]));
```

Comment:
I think there are several CV terms associated with "signal_unit", "detector", etc., that we are currently pushing into the MetaValues.  It would be best to use the CV terms if possible, however, I am not too familiar with the preferred way to add this information manually into the mzML file like we are doing.  

I think it might be best to contact Timo or Hannes about the best way of integrating this meta info.